tests/test_rdf.py

```diff
@@ -211,6 +211,48 @@ def test_add_rdf(local_model):
     assert named_attributes[0]['value'] == '2.5' or named_attributes[1]['value'] == '2.5'
 
 
-def test_get_unique_cmeta_id(simple_ode_model):
-    assert simple_ode_model.get_unique_cmeta_id('already_unique_id') == 'already_unique_id'
-    assert simple_ode_model.get_unique_cmeta_id('time') == 'time_'
+def test_add_cmeta_id():
+    """ Tests Model.add_cmeta_id(). """
+
+    model = shared.load_model('test_simple_odes')
+
+    # Test on variable without a cmeta id
+    v = model.get_symbol_by_name('circle_x$x')
+    assert v.rdf_identity is None
+    model.add_cmeta_id(v)
+    assert v.rdf_identity is not None
+
+    # Test on variable with a cmeta id
+    v = model.get_symbol_by_ontology_term((shared.OXMETA, 'time'))
+    rid = v.rdf_identity
+    assert rid is not None
+    model.add_cmeta_id(v)
+    assert v.rdf_identity == rid
+
+    # Test on variable with a name that's already in use as a cmeta id
+    assert model.has_cmeta_id('time')
+    v = model.add_variable('time', 'second')
+    model.add_cmeta_id(v)
+
+
+def test_transfer_cmeta_id():
+    """ Tests Model.transfer_cmeta_id(). """
+
+    model = shared.load_model('test_simple_odes')
+
+    # Legal move: from variable with a cmeta id to one without
+    v1 = model.get_symbol_by_ontology_term((shared.OXMETA, 'time'))
+    v2 = model.get_symbol_by_name('circle_x$x')
+    assert v1.rdf_identity is not None
+    assert v2.rdf_identity is None
+    model.transfer_cmeta_id(v1, v2)
+    assert v1.rdf_identity is None
```

Comment:
Interesting that this passed given my earlier comment! I guess because the property checks `_cmeta_id` first.

Reply:
Why shouldn't this pass?

Reply:
It should! I'd just assumed because self._rdf_identity is cached that it wouldn't, but rdf_identity has an extra check before accessing the cache which is why it does currently. That check won't be needed now.